# DETERMINING THE HIGHEST AND LOWEST HOUSE PRICES 

## Data loading

Load pandas library used to load the house prices dataset

In [2]:
import pandas as pd 

Load the dataset using pandas library and view the data types of the variables contained in the dataset

In [3]:
data = pd.read_csv("house_prices.csv")

In [4]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 261 entries, 0 to 260
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Date             261 non-null    object 
 1   Price (All)      261 non-null    int64  
 2   Change (All)     261 non-null    float64
 3   Price (New)      261 non-null    int64  
 4   Change (New)     261 non-null    float64
 5   Price (Modern)   261 non-null    int64  
 6   Change (Modern)  261 non-null    float64
 7   Price (Older)    261 non-null    int64  
 8   Change (Older)   261 non-null    float64
dtypes: float64(4), int64(4), object(1)
memory usage: 18.5+ KB


Head of the dataset displaying values from different variables in the dataset

In [5]:
data.head()

Date  Price (All)  Change (All)  Price (New)  ...  Price (Modern)  Change (Modern)  Price (Older)  Change (Older)
0  1952-11-01         1891           0.0         2107  ...            2020              0.0           1524             0.0
1  1953-02-01         1891           0.0         2107  ...            2002              0.0           1542             0.0
2  1953-05-01         1891           0.0         2107  ...            2002              0.0           1542             0.0
3  1953-08-01         1881           0.0         2117  ...            2002              0.0           1524             0.0
4  1953-11-01         1872          -1.0         2117  ...            1975             -2.2           1542             1.2

[5 rows x 9 columns]

## Data cleaning

Dropping all the values which are null from the dataset. Cleaning the data through dropping the null values increases the accuracy of the outcome of the analysis part.

In [6]:
data = data.dropna()

Dropping columns which are not usable for further analysis on the paper is crucial toward reducing the occurance of a mix-up during the analysis phase.

In [7]:
data = data.drop(columns=['Date','Change (All)','Change (New)','Price (Older)','Change (Modern)','Change (Older)'])

## Data transformation

The transformation is undertaken through arranging the data in a descending order based on the Price (Modern) column.  

In [17]:
data = data.sort_values(ascending=False, by=["Price (Modern)"])

In [18]:
data.head(10)

Price (All)  Price (New)  Price (Modern)
250       194258       194511          179358
248       189002       196136          174683
249       188566       192434          174053
247       188810       193409          173918
219       184131       190476          172398
217       175554       184424          172247
220       183959       192268          171753
246       186544       190051          171680
218       181810       190214          169967
216       172065       180964          169066

## Data analysis

The data analysis is undertaken through determining the highest and lowest prices of house available in the dataset.

In [11]:
a = data['Price (All)']


In [14]:
print("Highest house price is:",max(a))

Highest house price is: 211672


In [15]:
print("Lowest house price is:",min(a))

Lowest house price is: 1853


According to the findings above the highest price of a house is $211,672


In [ ]:
The lowest price for houses based on the dataset is $1,853